manifests/remote_file.pp

```diff
@@ -0,0 +1,20 @@
+# Downloads a file from a URL to a local file given by the title
+define foreman_proxy::remote_file(
+  $remote_location,
+  $mode='0644',
+) {
+  $parent = dirname($title)
+  File <| title == $parent |>
+  -> exec { "mkdir -p ${parent}":
+    path    => ['/bin', '/usr/bin'],
+    creates => $parent,
+  }
+  -> exec {"retrieve_${title}":
+    command => "/usr/bin/curl -s ${remote_location} -o ${title}",
```

Comment:
want to add `--connect-timeout foo` since your setting `timeout => 0,` below?

Reply:
File doesn't expose a timeout setting so that may be tricky.